hazelcast/src/main/java/com/hazelcast/datalink/Resource.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink;
+
+import javax.annotation.Nonnull;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * DataLink Resource is an object for which a mapping can be created.
+ * <p>
+ * For JDBC it is the list of tables and views, for Kafka the list
+ * of topics, and for a filesystem the list of files etc.
+ */
+public class Resource {
```

Comment:
There is already a ResourceConfig in Jet and AFAIK it has nothing to do with this Resource. Perhaps rename it to DatalinkResource to prevent such confusion and any conflicts with other Resource usages like javax.annotation.Resource?

Reply:
`5d104102cfc`